pkg/profilestore/profilecolumnstore.go

```diff
@@ -139,10 +154,98 @@ func (s *ProfileColumnStore) WriteRaw(ctx context.Context, req *profilestorepb.W
 			}
 
 			if err := ingester.Ingest(ctx, ls, p, req.Normalized); err != nil {
-				return nil, status.Errorf(codes.Internal, "failed to ingest profile: %v", err)
+				return status.Errorf(codes.Internal, "failed to ingest profile: %v", err)
 			}
 		}
 	}
 
+	return nil
+}
+
+func (s *ProfileColumnStore) updateAgents(ip string, ag agent) {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	s.agents[ip] = ag
```

Comment:
When updating the agents it probably makes sense to also clean up "old" agents that don't push anymore.
It's probably best to do this in a separate goroutine eventually, but for now, I think we can start cleaning them up here.
Essentially something like this:
```suggestion
	s.agents[ip] = ag
	
	for ip, a := range s.agents {
		if a.lastPush.Before(time.Now().Add(-5*time.Minute) {
			delete(s.agents, ip)
		}
	}
```
_I've written this code here on GitHub so it might not be 100% correct._

Already on https://demo.parca.dev I can see this becoming a problem very quickly, where there are always 4 agents at a time right now and we might deploy or restart them a couple of times daily.

Reply:
Ahh, i forgot to handle this. Thanks for the review and pointing this out! Done.